pages/api/calculateEligibility.ts

```diff
@@ -10,6 +10,19 @@ import {
   ResultOptions,
 } from '../../utils/api/types'
 
+type formField = {
+  fieldId: string
+  fieldLabel: string
+  fieldRequired: boolean
+  fieldPlaceholder: string
+  fieldType: 'text' | 'number' | 'email' | 'radio' | 'checkbox'
+}
+
+type allFieldProposal = {
+  // ..other data ...
+  fields: formField[]
+}
```

Comment:
First pass at what I was thinking, where `allFields` would instead just be exposed as `fields` and we could parse that information into a component factory to create the form.

From there we stuff user inputs into a context (might need a state management library to stop unnecessary re-renders) and submit updates onChange.